diesel/src/query_builder/upsert/on_conflict_target_decorations.rs

```diff
@@ -0,0 +1,69 @@
+use crate::backend::{Backend, SupportsOnConflictClause, SupportsOnConflictTargetDecorations};
+use crate::expression::Expression;
+use crate::query_builder::upsert::on_conflict_target::{ConflictTarget, NoConflictTarget};
+use crate::query_builder::where_clause::{NoWhereClause, WhereAnd, WhereClause};
+use crate::query_builder::{AstPass, QueryFragment, QueryResult};
+use crate::sql_types::BoolOrNullableBool;
+
+pub trait UndecoratedConflictTarget {}
+
+impl UndecoratedConflictTarget for NoConflictTarget {}
+impl<T> UndecoratedConflictTarget for ConflictTarget<T> {}
+
+/// Interface to add information to conflict targets.
+/// Designed to be open for further additions to conflict targets like constraints
+pub trait DecoratableTarget<P> {
```

Comment:
Is there any reason why this needs to be a separate trait and we cannot reuse `FilterDsl` here?

Reply:
Thanks for the review @weiznich!

My initial thought also was to use `FilterDsl`. However, I created a new trait for 2 reasons
1. For me it is semantically different to add a `WHERE` clause on a `SELECT` or `UPDATE` statement which filters the statement as a _whole_. In contrast the `WHERE` clause added to a conflict target specifies the _on conflict_ policy. These statements are usually chained together so to me it made sense to indicate that the `WHERE` clause groups with the conflict target. 
2. I designed the `DecoratableTarget` such that it easily allows for further specification of the _on conflict_ policy such as adding constraint names as indicated in the PostgreSQL docs on insert. 

does that make sense to you?

Reply:
That sounds reasonable :+1: